test/test_linalg.py

```diff
@@ -994,6 +994,58 @@ def get_value_type(dtype):
                         with self.assertRaisesRegex(RuntimeError, r'provided dtype must match dtype of result'):
                             torch.linalg.norm(input, ord=ord, keepdim=keepdim, dtype=dtype, out=result)
 
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    def test_vector_norm(self, device, dtype):
+        # This test compares torch.linalg.vector_norm's output with
+        # torch.linalg.norm given a flattened tensor
+        ord_vector = [0, 0.5, 1, 2, 3, inf, -0.5, -1, -2, -3, -inf, None]
+        test_cases = [
+            # input size, ords that should cause error
+            ((10, ), []),
+            ((4, 5), []),
+            ((3, 4, 5), []),
+            ((3, 4, 5, 6, 7), []),
+
+            # Infinite orders cause an error for degenerate inputs because
+            # min/max cannot be called on degenerate inputs
+            ((0, ), [inf, -inf]),
+            ((0, 10), [inf, -inf]),
+            ((10, 0), [inf, -inf]),
+            ((0, 0), [inf, -inf]),
+            ((10, 0, 10), [inf, -inf]),
+            ((0, 0, 0), [inf, -inf]),
+        ]
+
+        def vector_norm_reference(input, ord, keepdim=False, dtype=None):
+            input_flat = input.flatten(0, -1)
+            result = torch.linalg.norm(input_flat, ord, keepdim=keepdim, dtype=dtype)
+            if keepdim:
+                result = result.reshape([1] * input.dim())
+            return result
+
+        def run_test_case(input, ord, keepdim, norm_dtype, should_error):
+            msg = f'input.size()={input.size()}, ord={ord}, keepdim={keepdim}, dtype={dtype}, norm_dtype={norm_dtype}'
+            if should_error:
+                with self.assertRaises(RuntimeError):
+                    vector_norm_reference(input, ord, keepdim=keepdim)
+                with self.assertRaises(RuntimeError):
+                    torch.linalg.vector_norm(input, ord, keepdim=keepdim)
+            else:
+                result_reference = vector_norm_reference(input, ord, keepdim=keepdim, dtype=norm_dtype)
+                result = torch.linalg.vector_norm(input, ord, keepdim=keepdim, dtype=norm_dtype)
+                self.assertEqual(result, result_reference, msg=msg)
+
+        norm_dtypes = [None, torch.float, torch.double, torch.cfloat, torch.cdouble]
+
+        for (input_size, error_ords), ord, keepdim, norm_dtype in product(test_cases, ord_vector, [True, False], norm_dtypes):
```

Comment:
This isn't testing the dim(s?) argument?

Reply:
I've added dim arg cases to this test